app/views/posts/_editor.haml

```diff
@@ -10,7 +11,14 @@
       #icon-overlay
     #current-icon-dropdown
       - if item.character
-        = select_tag :icon_dropdown, options_for_select(item.character.icons.map{|i| [i.keyword, i.id]}, item.icon_id), prompt: "No Icon"
+        - if current_user.icon_picker_grouping? && item.character.galleries.count > 1
+          - icons = character_galleries.map(&:ordered_icons).flatten.map{|i| [i.keyword, i.id]}
+        - else
+          - icons = item.character.icons.map{|i| [i.keyword, i.id]}
+        - unless icons.present?
+          - icon = item.character.icon
+          - icons = [[icon.keyword, icon.id]] unless icon.nil?
+        = select_tag :icon_dropdown, options_for_select(icons, item.icon_id), prompt: "No Icon"
```

Comment:
This is the same logic as `CharacterPresenter` uses for galleries (and then the same logic as the Javascript to produce HTML from it), so maybe they should be unified somehow.